File: sample/src/main/java/vn/tiki/sample/util/TextViews.java
package vn.tiki.sample.util;

import android.text.Spanned;
import android.widget.TextView;
import java.text.DecimalFormat;

public final class TextViews {
  private static final DecimalFormat DECIMAL_FORMAT = new DecimalFormat("#,###,###,###");

  public static void setHtml(TextView textView, String htmlContent) {
    final Spanned html = Strings.Companion.toHtml(htmlContent);
    textView.setText(html);
  }

  public static void setPrice(TextView textView, float price) {
    final CharSequence text = formatPrice(price);
    textView.setText(text);
  }

  private TextViews() {
    throw new InstantiationError();
  }

  private static CharSequence formatPrice(float price) {
    final String result = DECIMAL_FORMAT.format(price);
    return String.format("$%s", result);
  }
}

File: viewHoldersDemo/src/main/java/vn/tale/viewholdersdemo/util/TextViews.kt
package vn.tale.viewholdersdemo.util

import android.widget.TextView
import java.text.DecimalFormat

private val decimalFormat = DecimalFormat("#,###,###,###")

fun Float.toDecimal(): String {
  val result = decimalFormat.format(toDouble())
  return String.format("$%s", result)
}

fun TextView.setPrice(price: Float) {
  text = price.toDecimal()
}
